### PR TITLE
Add default product source in Courses Api Data Loader

### PR DESCRIPTION
**Pull request by zawan-ila**
## Testing Instructions

1. Checkout this branch
2. Create a course in studio
3. Run RCM
4. Verify that the course you created in studio is visible in discovery and has the default product source assigned.

### CODE REVIEW

File: course_discovery/apps/course_metadata/tests/test_signals.py [hunk 1]
```diff
@@ -746,6 +746,7 @@ def setUp(self):
             hidden=False,
         )
         self.partner = PartnerFactory(id=settings.DEFAULT_PARTNER_ID)
+        _ = factories.SourceFactory(slug=settings.DEFAULT_PRODUCT_SOURCE_SLUG)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/test_signals.py]: No need for storing the output in `_`

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 2]
```diff
@@ -246,6 +253,52 @@ def test_ingest_verified_deadline(self, mock_push_to_ecomm):
...
+        assert CourseRun.objects.count() == len(api_data)
+        courses_created_count = Course.objects.count()
+        assert courses_created_count > 0
+
+        # Verify that all created courses have the default product source
+        product_source_slugs = (
+            Course.objects.select_related('product_source')
+            .values_list('product_source__slug', flat=True)
+        )
+        assert set(product_source_slugs) == {settings.DEFAULT_PRODUCT_SOURCE_SLUG}
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py]: You can handle the default source handle in an existing test. You can have this test case to mention that source is not altered if the course is being updated.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 3]
```diff
@@ -217,6 +217,10 @@ def get_or_create_course(self, body):
 
             course.authoring_organizations.add(organization)
 
+            # Set the default product source
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: there is no need for this comment. The code ahead is self-explanatory.

File: course_discovery/apps/course_metadata/data_loaders/api.py [hunk 4]
```diff
@@ -217,6 +217,10 @@ def get_or_create_course(self, body):
...
+            course.product_source = Source.objects.get(slug=settings.DEFAULT_PRODUCT_SOURCE_SLUG)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/api.py]: When running RCM, the default source will be fetched again & again. Ideally, get this value in init and assign the obj here when a course is being created.

File: course_discovery/apps/course_metadata/tests/test_signals.py [hunk 5]
```diff
@@ -730,6 +730,11 @@ def test_authoring_organizations_changed(self):
 
 
 class TestCourseDataUpdateSignal(TestCase):
+    @classmethod
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/tests/test_signals.py:743]: nit: add a line here, for readability purpose.

**Review by DawoudSheraz — APPROVED** [review]
I have not tested it out on the local. The changes are plain & simple 👍🏽

**Review by AliAdnanSohail — APPROVED** [review]
LGTM

**Pull request merged**